evm/v0.5/contracts/dev/PrepaidAggregator.sol

```diff
@@ -0,0 +1,242 @@
+pragma solidity 0.5.0;
+
+import "../Median.sol";
+import "../vendor/Ownable.sol";
+import "../vendor/SafeMath.sol";
+import "./SafeMath128.sol";
+import "../interfaces/LinkTokenInterface.sol";
+
+/**
+ * @title The PrepaidAggregator handles aggregating data pushed in from off-chain.
+ */
+contract PrepaidAggregator is Ownable {
+  using SafeMath for uint256;
+  using SafeMath128 for uint128;
+
+  struct OracleStatus {
+    bool enabled;
+    uint128 withdrawable;
+    uint256 lastReportedRound;
+  }
+
+  struct Round {
+    uint64 maxAnswers;
+    uint64 minAnswers;
+    uint128 paymentAmount;
+    int256[] answers;
+  }
+
+  int256 public currentAnswer;
+  uint256 public currentRound;
+  uint128 public paymentAmount;
+  uint64 public oracleCount;
+  uint64 public maxAnswerCount;
+  uint64 public minAnswerCount;
+  uint256 public updatedHeight;
+  uint128 public availableFunds;
+  uint128 public allocatedFunds;
+
+  LinkTokenInterface private LINK;
+  mapping(address => OracleStatus) private oracles;
+  mapping(uint256 => Round) private rounds;
+
+  event NewRound(uint256 indexed number, address indexed startedBy);
+  event AnswerUpdated(int256 indexed current, uint256 indexed round);
+  event AvailableFundsUpdated(uint256 indexed amount);
+
+  constructor(address _link, uint128 _paymentAmount) public {
+    LINK = LinkTokenInterface(_link);
+    setPaymentAmount(_paymentAmount);
+  }
+
+  function updateAnswer(uint256 _round, int256 _answer)
+    public
+    ensureValidRoundId(_round)
+    validateOracleRound(_round)
+  {
+    startNewRound(_round);
+    recordAnswer(_answer, _round);
+    updateRoundAnswer(_round);
```

Comment:
Wouldn't the first suggestion be configurable by setting `minAnswerCount` and `maxAnswerCount` to the same thing?

Reply:
Seems like for the second suggestion it might be worth making a second contract. What do you think?

Reply:
Yeah, the second suggestion is not for this review, just an observation that if there are going to be repeated updates a more elaborate data structure is going to be more efficient.

Setting `minAnswerCount == maxAnswerCount` prevents the repeated updates, but if that's the solution, it might be best to merge them into a single variable.